processor/src/operations/io_ops.rs

```diff
@@ -43,6 +43,29 @@ impl Process {
         Ok(())
     }
 
+    /// Loads an element from the specified memory address onto the stack.
+    ///
+    /// The operation works as follows:
+    /// - The memory address is popped off the stack.
+    /// - A word is retrieved from memory at the specified address. The memory is always
+    ///   initialized to ZEROs, and thus, if the specified address has never been written to,
+    ///   four ZERO elements are returned.
+    /// - The top element of the stack is overwritten with the first element of the word retried 
+    ///   from memory.
+    ///
+    /// Thus, the net result of the operation is that the stack is shifted left by one item.
+    pub(super) fn op_mload(&mut self) -> Result<(), ExecutionError> {
+        // get the address from the stack and read the word from memory
+        let addr = self.stack.get(0);
+        let word = self.memory.read(addr);
+
+        // update the stack state
+        self.stack.set(0, word[0]);
+        self.stack.shift_left(2);
```

Comment:
I'm not sure this is correct. We are overwriting the element at the top of the stack, so, the rest of the stack doesn't need to be shifted.

Reply:
But during the `MSTORE` operation, we have to pop the address from the top of the stack. Don't we need to shift it left for that? Do I understand the stack transition correctly?
```
[addr, value, a, b, ... ] -> [value, a, b, ... ]
```
I realized that this block of code can be replaced with  `self.stack.shift_left(1);`, but the output will remain the same